### PR TITLE
User activation: fixes and improvements to utils

### PR DESCRIPTION
**Pull request by marcoscaceres**
* Add attachIframe() utility method
* Attempts to use test_driver.consume_user_activation if available
* Adds a few additional JSDoc comments

### CODE REVIEW

File: html/user-activation/resources/utils.js [hunk 1]
```diff
@@ -1,37 +1,56 @@
...
+ * @param {Function} f
+ * @param {number} num_frames
+ */
 function delayByFrames(f, num_frames) {
   function recurse(depth) {
-    if (depth == 0)
-      f();
-    else
-      requestAnimationFrame(() => recurse(depth-1));
+    if (depth == 0) f();
```
Comment (annevk) [inline html/user-activation/resources/utils.js:9]: Not a big fan of this style.

**Review by annevk — APPROVED** [review]

File: html/user-activation/resources/utils.js (hunk 1 above)
Comment (marcoscaceres) [reply html/user-activation/resources/utils.js:9]: yeah, I'm not a huge fan either... was just keeping the original. I'll see about adding { } in a followup.

**Comment by jgraham** [comment]
The changes to `consumeTransientActivation` appear to have regressed [activation-trigger-pointerevent.html](https://wpt.fyi/results/html/user-activation/activation-trigger-pointerevent.html%3Fmouse?label=master&label=experimental&product=chrome&product=firefox&product=safari&aligned&view=interop&q=label%3Ainterop-2023-events) which is part of interop. Given that I'm going to create a PR to revert this change until the bug is fixed.

**Comment by mustaqahmed** [comment]
I just spotted the scary breakage in the [Interop 2023 dashboard](https://wpt.fyi/results/html/user-activation/activation-trigger-pointerevent.html%3Fmouse?label=master&label=experimental&product=chrome&product=firefox&product=safari&aligned&view=interop&q=label%3Ainterop-2023-events).  Thanks @jgraham for the super-quick revert.

**Comment by marcoscaceres** [comment]
Will check what happened...

**Pull request merged**